src/dialog/tests/unit/Dialog.ts

```diff
@@ -19,84 +37,119 @@ registerSuite({
+
+		]);
+		dialog.expectRender(expected);
 	},
 
 	'Render correct children'() {
-		const dialog = new Dialog();
-		dialog.__setProperties__({
+		dialog.setProperties({
 			enterAnimation: 'enter',
 			exitAnimation: 'exit',
 			role: 'dialog'
 		});
-		let vnode = <VNode> dialog.__render__();
-		assert.strictEqual(vnode.vnodeSelector, 'div', 'tagname should be div');
-		assert.property(vnode.properties!.classes!, css.root);
+		let vnode = <HNode> dialog.getRender();
 		assert.lengthOf(vnode.children, 0);
 
-		dialog.__setProperties__({
+		dialog.setProperties({
 			open: true,
 			underlay: true,
 			role: 'dialog'
 		});
-		vnode = <VNode> dialog.__render__();
+		vnode = <HNode> dialog.getRender();
 		assert.lengthOf(vnode.children, 2);
 	},
 
 	onRequestClose() {
-		const dialog = new Dialog();
-		dialog.__setProperties__({
+		let called = false;
+		dialog.setProperties({
 			open: true,
-			onRequestClose: () => {
-				dialog.__setProperties__({ open: false });
-			}
+			closeable: true,
+			onRequestClose: () => called = true
 		});
-		(<any> dialog)._onCloseClick();
-
-		assert.isFalse(dialog.properties.open, 'onRequestClose should be called when close button is clicked');
+		dialog.sendEvent('click', { selector: 'button'});
+		assert.isTrue(called, 'onRequestClose should be called when close button is clicked');
 	},
 
 	onOpen() {
 		let called = false;
 
-		const dialog = new Dialog();
-		dialog.__setProperties__({
+		dialog.setProperties({
 			open: true,
 			onOpen: () => {
 				called = true;
 			}
 		});
-		<VNode> dialog.__render__();
+		dialog.getRender(); // just to trigger a `_invalidate()`
 
 		assert.isTrue(called, 'onOpen should be called');
 	},
 
 	modal() {
-		const dialog = new Dialog();
-		dialog.__setProperties__({
+		let called = false;
+		dialog.setProperties({
 			open: true,
 			modal: true,
-			onRequestClose: () => {
-				dialog.__setProperties__({ open: false });
-			}
+			onRequestClose: () => called = true
 		});
-		(<any> dialog)._onUnderlayClick();
-
-		assert.isTrue(dialog.properties.open, 'dialog should stay open when underlay is clicked and modal is true');
+		dialog.sendEvent('click', { key: 'underlay' } );
+		assert.isFalse(called, 'onRequestClose should not be called when underlay is clicked and modal is true');
 
-		dialog.__setProperties__({ modal: false });
-		(<any> dialog)._onUnderlayClick();
-
-		assert.isUndefined(dialog.properties.open, 'dialog should close if underlay is clicked and modal is false');
+		dialog.setProperties({
+			open: true,
+			modal: false,
+			onRequestClose: () => called = true
+		});
+		dialog.getRender(); // just to trigger a `_invalidate()`
+		dialog.sendEvent('click', { key: 'underlay' } );
+		assert.isTrue(called, 'onRequestClose should be called if underlay is clicked and modal is false');
 	},
 
 	closeable() {
+		// this test can not be converted using `test-extras` because there's no way to use `callListener` or `sendEvent` without the close button being rendered
```

Comment:
Can you remove this comment please.
Can you not sent closeable to false and then check to see if the close icon is rendered via the `expectRender` call?